src/components/webview/MainProvider.ts

```diff
@@ -0,0 +1,68 @@
+import {
+	WebviewViewProvider,
+	WebviewView,
+	Uri,
+	ExtensionContext,
+} from 'vscode';
+
+import { WebviewResolver } from './WebviewResolver';
+import { CommunityProvider } from './CommunityProvider';
+import { CampaignManagerProvider } from './CampaignManagerProvider';
+import { FileExplorerProvider } from './FileExplorerProvider';
+import { CodemodListPanelProvider } from './CodemodListProvider';
+
+export class MainViewProvider implements WebviewViewProvider {
+	__view: WebviewView | null = null;
+	__extensionPath: Uri;
+	__webviewResolver: WebviewResolver;
+
+	constructor(
+		context: ExtensionContext,
+		private readonly __community: CommunityProvider,
+		private readonly __codemodRuns: CampaignManagerProvider,
+		private readonly __fileExplorer: FileExplorerProvider,
+		private readonly __codemodList: CodemodListPanelProvider,
+	) {
+		this.__extensionPath = context.extensionUri;
+		this.__webviewResolver = new WebviewResolver(this.__extensionPath);
+	}
+
+	public resolveWebviewView(webviewView: WebviewView): void | Thenable<void> {
+		if (!webviewView.webview) {
+			return;
+		}
+
+		this.__resolveWebview(webviewView);
+
+		this.__view = webviewView;
+
+		this.__community.resolveWebviewView(webviewView);
```

Comment:
can you tell me how this works? There is just one HTML in the end, am I right?

Reply:
main webview provider resolves webview, attaches HTML to it, and passes it down to classes tthat we previously had.
method name is slightly misleading, i will rename it to `setWebview`.